cmd/nerdctl/restart_linux_test.go

```diff
@@ -69,3 +70,29 @@ func TestRestartPIDContainer(t *testing.T) {
 
 	assert.Equal(t, baseOutput, sharedOutput)
 }
+
+func TestRestartWithTime(t *testing.T) {
+	t.Parallel()
+	base := testutil.NewBase(t)
+	tID := testutil.Identifier(t)
+
+	// here we do not use nginx-alpine because it restarts immediately after
+	// receiving the SIGTERM signal with ignoring the time specified
+	base.Cmd("run", "-d", "--name", tID, testutil.AlpineImage, "sleep", "infinity").AssertOK()
+	defer base.Cmd("rm", "-f", tID).AssertOK()
+	base.EnsureContainerStarted(tID)
+
+	inspect := base.InspectContainer(tID)
+	pid := inspect.State.Pid
+
+	timeStart := time.Now()
+	base.Cmd("restart", "-t", "5", tID).AssertOK()
+	base.EnsureContainerStarted(tID)
+	timeEnd := time.Now()
+
+	newInspect := base.InspectContainer(tID)
+	newPid := newInspect.State.Pid
+	assert.Assert(t, pid != newPid)
+	// The wait time cost may float within +1 second
+	assert.Assert(t, timeEnd.Sub(timeStart).Round(time.Second) <= time.Second*6)
```

Comment:
You have to ensure that stop took at least 5s.

Reply:
SGTM